Source/WarpX.cpp

```diff
@@ -959,12 +960,24 @@ WarpX::AllocLevelMFs (int lev, const BoxArray& ba, const DistributionMapping& dm
 #elif (AMREX_SPACEDIM == 2)
         RealVect dx_vect(dx[0], dx[2]);
 #endif
+
+        // Check whether the option periodic, single box is valid here
+        if (fft_periodic_single_box) {
+            AMREX_ALWAYS_ASSERT_WITH_MESSAGE( geom[0].isAllPeriodic() && ba.size()==1 && lev==0,
+            "The option `psatd.periodic_single_box_fft` can only be used for a periodic domain, decomposed in a single box.");
+        }
+
         // Get the cell-centered box, with guard cells
         BoxArray realspace_ba = ba;  // Copy box
-        realspace_ba.enclosedCells().grow(ngE); // cell-centered + guard cells
+        realspace_ba.enclosedCells(); // cell-centered
+        if ( fft_periodic_single_box == false ) {
+            realspace_ba.grow(ngE); // add guard cells
+        }
         // Define spectral solver
+        bool const pml=false;
```

Comment:
Could we also ASSERT that `warpx.do_pml = false` so we can pass it to the function call below, instead of re-defining a variable?

Reply:
Actually, sorry the code is not very clear here.
When we use PML, we have two types of MultiFabs:
- the MultiFabs for regular physical cells
- the MultiFabs for the PML cells.
Each of these MultiFabs have their spectral solver ; however, these spectral solver solve different equations depending on whether they are applied to regular physical cells or PML cells. In order for the spectral solver to know which type of equation to solve, we pass `pml=true` or `pml=false` when initializing the spectral solver.
The above code corresponds to the regular physical cells ; even when `do_pml=1`, we should still pass `pml=false` here.
The code for the spectral solver applied to PML cells is here: 
https://github.com/ECP-WarpX/WarpX/blob/master/Source/BoundaryConditions/PML.cpp#L477
(that code was not modify to support `periodic_single_box` because this option is incompatible with PML).

Does the above make sense? I am not sure how to convey the above ideas concisely in the code, in the form comments... If you have any suggestion, please feel free to propose them here.

Reply:
Thanks for the clarification. I think it is fine this way then.